execution/binary/vector.go

```diff
@@ -163,9 +169,26 @@ func (o *vectorOperator) Next(ctx context.Context) ([]model.StepVector, error) {
 	batch := o.pool.GetVectorBatch()
 	for i, vector := range lhs {
 		if i < len(rhs) {
-			step := o.table.execBinaryOperation(lhs[i], rhs[i])
-			batch = append(batch, step)
-			o.rhs.GetPool().PutStepVector(rhs[i])
+			step, err := o.table.execBinaryOperation(lhs[i], rhs[i])
+			if err == nil {
+				batch = append(batch, step)
+				o.rhs.GetPool().PutStepVector(rhs[i])
+				continue
+			}
+
+			var sampleID, duplicateSampleID labels.Labels
+			switch err.side {
+			case lhBinOpSide:
+				sampleID = o.lhSampleIDs[err.sampleID]
+				duplicateSampleID = o.lhSampleIDs[err.duplicateSampleID]
+			case rhBinOpSide:
+				sampleID = o.rhSampleIDs[err.sampleID]
+				duplicateSampleID = o.rhSampleIDs[err.duplicateSampleID]
+			}
```

Comment:
```suggestion
			sampleID = o.lhSampleIDs[err.sampleID]
			duplicateSampleID = o.lhSampleIDs[err.duplicateSampleID]
			if err.side == rhBinOpSide {
				sampleID = o.rhSampleIDs[err.sampleID]
				duplicateSampleID = o.rhSampleIDs[err.duplicateSampleID]
			}
```

Reply:
I did have this before, but I find the switch statement a bit more explicit and easier to read

Reply:
Yeah I was thinking in "left-right" terms, but you're right, the switch is better for readability here!